src/passes/transform/clock_gate.cpp

```diff
@@ -74,26 +85,40 @@ bool ClockGate::runOnModule(Module *m) {
   std::vector<CEInfo> to_replace;
   for (auto ipair : def->getInstances()) {
     auto inst = ipair.second;
-    if (instance_of(inst, "coreir", "reg")) {
+    bool is_coreir_reg = instance_of(inst, "coreir", "reg");
+    bool is_coreir_reg_arst = instance_of(inst, "coreir", "reg_arst");
+    if (is_coreir_reg || is_coreir_reg_arst) {
+      // NYI clock being negedge
+      if (!inst->getModArgs().at("clk_posedge")->get<bool>()) {
+        continue;
+      }
```

Comment:
If we're not supporting negedge reset yet, we'll need a similar check here.

Reply:
Fixed